### PR TITLE
Jest config

### PR DESCRIPTION
**Pull request by Jayakrishnan-MS**
Added unit test

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@modusoperandi/licit-glossary",
-  "version": "0.0.2",
+  "version": "0.0.7",
```
Comment (Tezra) [inline package.json:3]: Why .7?

File: src/glossaryCommand.test.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      GlossaryListUI,
+      glossaryCmd.createGlossaryObject(view),
+      {
+        modal: true,
+        IsChildDialog: false,
+        autoDismiss: false,
+      }
+    );
+    glossaryCmd._isGlossary =false;
+    glossaryCmd.deleteGlossaryNode(view.state,'term');
```
Comment (Tezra) [inline src/glossaryCommand.test.ts:72]: it without expect

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@modusoperandi/licit-glossary",
-  "version": "0.0.3",
+  "version": "0.0.7",
```
Comment (Tezra) [inline package.json:3]: Why .7?

File: src/glossaryCommand.test.ts (hunk 2 above)
Comment (Tezra) [inline src/glossaryCommand.test.ts:72]: it without expect

File: src/glossaryListUI.test.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+import { EditorRuntime, Glossary } from './types';
+jest.mock('./types', () => ({
+    getGlossary: jest.fn(),
+}));
+Enzyme.configure({ adapter: new Adapter() });
+describe('GlossaryListUI component init', () => {
+    beforeEach(() => {
+
+
+        // Set up the mock implementation of getGlossary
```
Comment (Tezra) [inline src/glossaryListUI.test.tsx:14]: unused

File: src/glossaryListUI.test.tsx [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        const wrapper = shallow(<GlossaryListUI {...gPropo} />);
+        const listUIRender = wrapper.instance();
+        expect(listUIRender).toBeDefined();
+    });
+
+    it('should call updateState method with null param', () => {
+        const wrapper = shallow(<GlossaryListUI {...gPropo} />);
+        const listUIRender = wrapper.instance();
+        const res = null;
+        listUIRender.updateState(res);
```
Comment (Tezra) [inline src/glossaryListUI.test.tsx:87]: it without expect (5 instances in this file)

File: src/glossaryView.test.ts [hunk 6]
```diff
@@ -0,0 +1,409 @@
...
+        const clickEvent2 = new MouseEvent('mouseclick', {
+            clientX: 280,
+            clientY: 125,
+        });
+        cView.setContentRight(clickEvent2, errorinfodiv, tooltip, ttContent);
+        const getNodePosEx = jest.spyOn(cView, 'getNodePosEx');
+        getNodePosEx.mockReturnValue(12);
+        // Call the close function
+        cView.close();
+        cView.setContentRight(clickEvent, errorinfodiv, tooltip, ttContent);
```
Comment (Tezra) [inline src/glossaryView.test.ts:162]: it without expect X5

File: src/index.test.ts [hunk 7]
```diff
@@ -1,37 +1,281 @@
...
+        {
+          state: state,
+        }
+      );
+      const gView = new GlossaryView(
+        view.state.doc.nodeAt(6),
+        view,
+        undefined as any
+      );
+      gView.createGlossaryObject(view, 1);
```
Comment (Tezra) [inline src/index.test.ts:79]: it without expect

File: src/index.test.ts [hunk 8]
```diff
@@ -1,37 +1,281 @@
...
+        glossaryCmd.createGlossaryObject(view),
+        {
+          modal: true,
+          IsChildDialog: false,
+          autoDismiss: false,
+        }
+      );
+      glossaryCmd.waitForUserInput(editor.state, undefined, view);
+      glossaryCmd._popUp = null;
+      glossaryCmd.waitForUserInput(editor.state, undefined, view);
```
Comment (Tezra) [inline src/index.test.ts:274]: it without expect

**Review by Tezra — CHANGES_REQUESTED** [review]

File: src/glossaryView.test.ts [hunk 9]
```diff
@@ -0,0 +1,409 @@
...
+            {
+                modal: true,
+                IsChildDialog: false,
+                autoDismiss: false,
+            }
+        );
+        cView.selectNode(mockEvent);
+        gView.runtime = {} as EditorRuntime;
+        gView.onEditGlossary(view);
+        cView.selectNode(event as MouseEvent);
```
Comment (Tezra) [inline src/glossaryView.test.ts:217]: it without expect

File: src/glossaryView.test.ts [hunk 10]
```diff
@@ -0,0 +1,409 @@
...
+        );
+        const targetElement = document.createElement('div');
+        const clickEvent = new MouseEvent('mouseclick', {
+            clientX: 281,
+            clientY: 125
+        });
+        targetElement.dispatchEvent(clickEvent);
+        cView.selectNode(clickEvent);
+        cView.selectNode(undefined as any);
+
```
Comment (Tezra) [inline src/glossaryView.test.ts:249]: it without expect

File: src/glossaryView.test.ts [hunk 11]
```diff
@@ -0,0 +1,409 @@
...
+                from: 0,
+                to: 9,
+                type: 1,
+                id: 1,
+                description: 'Test description',
+                term: 'term'
+            }
+        };
+        gView.updateGlossaryDetails({}, glossaryObj);
+        gView.updateGlossaryDetails(view, glossaryObj);
```
Comment (Tezra) [inline src/glossaryView.test.ts:328]: it without expect

File: src/glossaryView.test.ts [hunk 12]
```diff
@@ -0,0 +1,409 @@
...
+        gView._popUp = createPopUp(
+            GlossaryListUI,
+            glossaryCmd.createGlossaryObject(view),
+            {
+                modal: true,
+                IsChildDialog: false,
+                autoDismiss: false,
+            }
+        );
+        gView.destroyPopup();
```
Comment (Tezra) [inline src/glossaryView.test.ts:376]: it without expect

File: src/index.test.ts [hunk 13]
```diff
@@ -1,37 +1,289 @@
...
+      );
+      glossaryCmd.waitForUserInput(editor.state, undefined, view).then(
+        (value)=>{
+          expect(value).toBe(undefined);
+        } 
+      );
+      glossaryCmd._popUp = null;
+    glossaryCmd.waitForUserInput(editor.state, undefined, view).then(
+      (value)=>{
+        expect(value).toBe({});
```
Comment (Tezra) [inline src/index.test.ts:280]: desynced expect. Use async await instead of then.

File: src/index.test.ts [hunk 14]
```diff
@@ -1,37 +1,289 @@
...
+        glossaryCmd.createGlossaryObject(view),
+        {
+          modal: true,
+          IsChildDialog: false,
+          autoDismiss: false,
+        }
+      );
+      glossaryCmd.waitForUserInput(editor.state, undefined, view).then(
+        (value)=>{
+          expect(value).toBe(undefined);
```
Comment (Tezra) [inline src/index.test.ts:274]: desynced expect. Use async await instead of then.

**Review by Tezra — CHANGES_REQUESTED** [review]

**Review by rodgersgb — APPROVED** [review]

**Pull request merged**